tls/s2n_connection.c

```diff
@@ -1392,3 +1399,71 @@ int s2n_connection_set_keyshare_by_name_for_testing(struct s2n_connection *conn,
 
     POSIX_BAIL(S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
 }
+
+static int s2n_openssl_free(uint8_t** data) {
+    if (*data != NULL) {
+        OPENSSL_free(*data);
+        *data = NULL;
+    }
+    return S2N_SUCCESS;
+}
+
+static int s2n_sk_X509_pop_free(STACK_OF(X509) **cert_chain) {
+    if (*cert_chain != NULL) {
+        sk_X509_pop_free(*cert_chain, X509_free);
+        *cert_chain = NULL;
+    }
+    return S2N_SUCCESS;
+}
```

Comment:
Should these be static methods in s2n_connection.c? I'd suggest moving them (and their weird if/def imports) to the crypto folder, probably crypto/s2n_openssl_x509, although that's currently just a header.

Reply:
I plan on moving these static methods into crypto/ as a part of this issue: https://github.com/aws/s2n-tls/issues/2673

Reply:
Is that issue to handle ALL openssl crypto methods? Because that seems unlikely to be a fast followup. s2n_connection.c is already large (and also important) so we should try to keep it as clean as possible. What's the blocker to putting these methods in /crypto?